QuizApp/Presentation Layer/Quizzes/QuizzesPresenter.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+
+class QuizzesPresenter {
+    
+    private weak var router: AppRouter!
+    private var quizUseCase: QuizUseCase!
+    private var categories: [QuizCategory] = []
+    private var quizzes: [Quiz] = []
+    
+    init(router: AppRouter, quizUseCase: QuizUseCase) {
+        self.router = router
+        self.quizUseCase = quizUseCase
+    }
+    
+    func fetchData(completionHandler: @escaping () -> Void) {
+        quizUseCase.fetchData(completionHandler: { [self] quizzes in
+            self.quizzes = quizzes
+            categories = QuizCategory.allCases.filter { category in
+                quizzes.contains(where: { $0.category == category })
+            }
+            
+            completionHandler()
+        })
+    }
+    
+    func fetchData(filter: String) {
+        self.quizzes = quizUseCase.fetchData(filter: filter)
+        categories = QuizCategory.allCases.filter { category in
+            quizzes.contains(where: { $0.category == category })
+        }
+    }
+    
+    func funFact() -> Int {
+        quizzes.flatMap { $0.questions }.map { $0.question }.filter({ $0.contains("NBA") }).count
+    }
+    
+    func numberOfSections() -> Int {
+        categories.count
+    }
```

Comment:
In Swift, it's common to use computed properties instead of functions that don't receive any parameters.
```swift
var numberOfSections: Int {
         categories.count
}
```